lib/rspec/support/source/location.rb

```diff
@@ -0,0 +1,21 @@
+module RSpec
+  module Support
+    class Source
+      # @private
+      # Represents a source location of node or token.
+      Location = Struct.new(:line, :column) do
+        include Comparable
+
+        def self.location?(array)
+          array.is_a?(Array) && array.size == 2 && array.all? { |e| e.is_a?(Integer) }
+        end
+
+        def <=>(other)
+          line_comparison = (line <=> other.line)
+          return line_comparison unless line_comparison == 0
+          column <=> other.column
+        end
```

Comment:
It's needed here https://github.com/rspec/rspec-expectations/pull/987/commits/c074ca900790336204ae3e18e88c03ebf337ce24#diff-9b277082395427eb0159d15a3204096bR176.